python/semantic_kernel/connectors/ai/anthropic/services/anthropic_chat_completion.py

```diff
@@ -254,39 +278,157 @@ def _create_chat_message_content(
             inner_content=response,
             ai_model_id=self.ai_model_id,
             metadata=response_metadata,
-            role=AuthorRole(response.role),
+            role=AuthorRole.ASSISTANT,
             items=items,
             finish_reason=finish_reason,
         )
 
     def _create_streaming_chat_message_content(
         self,
-        stream_event: RawContentBlockDeltaEvent | RawMessageDeltaEvent,
-        content_block_idx: int,
-        role: str | None = None,
+        stream_event: TextEvent | ContentBlockStopEvent | RawMessageDeltaEvent,
         metadata: dict[str, Any] = {},
     ) -> StreamingChatMessageContent:
-        """Create a streaming chat message content object from a choice."""
-        text_content = ""
-
-        if stream_event.delta and hasattr(stream_event.delta, "text"):
-            text_content = stream_event.delta.text
-
-        items: list[STREAMING_ITEM_TYPES] = [StreamingTextContent(choice_index=content_block_idx, text=text_content)]
-
+        """Create a streaming chat message content object from a content block."""
+        items: list[STREAMING_ITEM_TYPES] = []
         finish_reason = None
-        if isinstance(stream_event, RawMessageDeltaEvent):
-            if stream_event.delta.stop_reason:
-                finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[stream_event.delta.stop_reason]
 
-            metadata["usage"].completion_tokens = stream_event.usage.output_tokens
+        if isinstance(stream_event, TextEvent):
+            items.append(StreamingTextContent(choice_index=0, text=stream_event.text))
+        elif (
+            isinstance(stream_event, ContentBlockStopEvent)
+            and hasattr(stream_event, "content_block")
+            and stream_event.content_block.type == "tool_use"
+        ):
+            tool_use_block = stream_event.content_block
+            items.append(
+                FunctionCallContent(
+                    id=tool_use_block.id,
+                    index=stream_event.index,
+                    name=tool_use_block.name,
+                    arguments=json.dumps(tool_use_block.input) if tool_use_block.input else None,
+                )
+            )
+        elif isinstance(stream_event, RawMessageDeltaEvent):
+            finish_reason = ANTHROPIC_TO_SEMANTIC_KERNEL_FINISH_REASON_MAP[str(stream_event.delta.stop_reason)]
+            metadata["usage"]["output_tokens"] = stream_event.usage.output_tokens
 
         return StreamingChatMessageContent(
-            choice_index=content_block_idx,
+            choice_index=0,
```

Comment:
Same question here -- we have `additive` operations that are based on `choice_index`. Won't that affect how two chunks could be combined?

Reply:
"Choices"/multiple messages isn't supported by Anthropic. Their message response contains content blocks, but this doesn't map to the "choices" behavior seen in other APIs